tools/MultiProjectValidator/AnalysisRules/DependencyMismatchRule.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.DotNet.ProjectModel;
+
+namespace MultiProjectValidator.AnalysisRules
+{
+    public class DependencyMismatchRule : IAnalysisRule
+    {
+        public AnalysisResult Evaluate(List<ProjectContext> projectContexts)
+        {
+            var targetGroupedContexts = GroupContextsByTarget(projectContexts);
+
+            var failureMessages = EvaluateTargetContextGroups(targetGroupedContexts);
+            var pass = failureMessages.Count == 0;
+
+            var result = new AnalysisResult(failureMessages, pass);
+            return result;
+        }
+
+        private List<string> EvaluateTargetContextGroups(Dictionary<string, List<ProjectContext>> targetGroupedContexts)
+        {
+            var failureMessages = new List<string>();
+
+            foreach (var target in targetGroupedContexts.Keys)
+            {
+                var targetContexts = targetGroupedContexts[target];
+
+                failureMessages.AddRange(EvaluateTargetContextGroup(targetContexts));
+            }
+
+            return failureMessages;
+        }
+
+        private List<string> EvaluateTargetContextGroup(List<ProjectContext> targetContexts)
+        {
+            var failedMessages = new List<string>();
+            var assemblyVersionMap = new Dictionary<string, string>();
+
+            foreach(var context in targetContexts)
+            {
+                var libraries = context.LibraryManager.GetLibraries();
+
+                foreach(var library in libraries)
+                {
+                    var name = library.Identity.Name;
+                    var version = library.Identity.Version.ToString();
+
+                    if (assemblyVersionMap.ContainsKey(name))
+                    {
+                        var existingVersion = assemblyVersionMap[name];
+                        if (!string.Equals(existingVersion, version, StringComparison.OrdinalIgnoreCase))
+                        {
+                            string message = 
```

Comment:
Yeah Michael you are right here, but it is somewhat intentional. Without making decisions about the conflicting dependencies involved (which would involve implementing some conflict resolution) this code can't know which projects are in the right or wrong. I don't think this code should make that decision either.

I do agree that the way the information is presented isn't optimal for a person coming along and trying to figure it out either. I really like your second example there, and will make that change.

Thanks! :smile: